core-api/src/test/java/com/optimizely/ab/optimizelyconfig/OptimizelyConfigTest.java

```diff
@@ -0,0 +1,115 @@
+        Map<String, OptimizelyExperiment> optimizelyExperimentMap = optimizelyConfig.getExperimentsMap();
+        assertEquals(optimizelyExperimentMap.size(), 13);
+
+        List<Experiment> experiments = getAllExperimentsFromDatafile();
+        experiments.forEach(experiment -> {
+            OptimizelyExperiment optimizelyExperiment = optimizelyExperimentMap.get(experiment.getKey());
+            assertNotNull(optimizelyExperiment);
+        });
+    }
+
+    @Test
+    public void shouldReturnAllFeatureFlag() throws Exception {
+        Map<String, OptimizelyFeature> optimizelyFeatureMap = optimizelyConfig.getFeaturesMap();
+        assertEquals(optimizelyFeatureMap.size(), 7);
+
+        projectConfig.getFeatureFlags().forEach(featureFlag -> {
+            OptimizelyFeature optimizelyFeature = optimizelyFeatureMap.get(featureFlag.getKey());
+            assertNotNull(optimizelyFeature);
+
+            Map<String, OptimizelyVariable> variablesMap = optimizelyFeatureMap.get(featureFlag.getKey()).getVariablesMap();
+            featureFlag.getVariables().forEach(variable -> {
+                OptimizelyVariable optimizelyVariable = variablesMap.get(variable.getKey());
+                assertNotNull(optimizelyFeature);
+            });
+        });
+    }
+
+    @Test
+    public void shouldCorrectlyMergeAllFeatureVariables() throws Exception {
+        List<FeatureFlag> featureFlags = projectConfig.getFeatureFlags();
+        Map<String, Experiment> datafileExperimentsMap = new HashMap<>();
+        getAllExperimentsFromDatafile().forEach(experiment ->
+            datafileExperimentsMap.put(experiment.getKey(), experiment)
+        );
+        featureFlags.forEach(featureFlag -> {
+            List<String> experimentIds = featureFlag.getExperimentIds();
+            experimentIds.forEach(experimentId -> {
+                String experimentKey = projectConfig.getExperimentIdMapping().get(experimentId).getKey();
+                OptimizelyExperiment experiment = optimizelyConfig.getExperimentsMap().get(experimentKey);
+                List<Variation> variations = datafileExperimentsMap.get(experimentKey).getVariations();
+                Map<String, OptimizelyVariation> variationsMap = experiment.getVariationsMap();
+                variations.forEach(variation ->
+                    featureFlag.getVariables().forEach(variable -> {
+                        OptimizelyVariable optimizelyVariable = variationsMap.get(variation.getKey()).getVariablesMap().get(variable.getKey());
+                        assertEquals(variable.getId(), optimizelyVariable.getId());
+                        assertEquals(variable.getKey(), optimizelyVariable.getKey());
+                        assertEquals(variable.getType().getVariableType().toLowerCase(), optimizelyVariable.getType());
+                        if (!variation.getFeatureEnabled()) {
+                            assertEquals(variable.getDefaultValue(), optimizelyVariable.getValue());
```

Comment:
Nit, but this is a bear to read. I don't think the `.forEach(...)` is buying us much except making is more difficult to follow.